### DISCUSSION TITLE
Limiting vehicle total distance between two Nodes identified by their visit order

### DISCUSSION THREAD

**Discussion by GuyBenhaim** [google/or-tools#2975] [Routing (and legacy CP) questions]
Hello,
I would like to limit the distance that a vehicle would travel between two nodes, which are characterized by the visit order the solver would assigns these (not the first and the last)
E.g. Travel distance between vehicles 3rd visited Node and its 7th visited node.

Does it mandate Dimension Dependent Dimension? (in other words, impossible ....)

Thx

**Comment by GuyBenhaim** [comment]
any tips?

**Comment by mslazynski** [comment]
Is there any reason why you want to identify nodes by their visit order?
It is even fairly difficult to assure how many nodes there will be in a single route...
If you just want to limit some dimension (e.g. distance) between  specific two nodes, you can try to add them as the pickup-delivery pair and then use "SetPickupToDeliveryLimitFunctionForPair" on the corresponding dimension.

**Comment by GuyBenhaim** [comment]
Looking quickly, this function cannot support what I wanted, since with regular Nodes I do not want to force the same vehicle, and therefore cannot guarantees it will be (like for P&D).
Still, **a neat function** that might come useful. Thx!
